### PR TITLE
add PostgreSQL 13 storage check

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: definitions/checks/disk/available_space_postgresql13.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+module Checks
+  module Disk
+    class AvailableSpacePostgresql13 < ForemanMaintain::Check
+      metadata do
+        label :available_space_for_postgresql13
+        description 'Check to make sure PostgreSQL 13 work directory has enough space for upgrade'
+        confine do
+          (feature(:foreman_database) || feature(:candlepin_database) || feature(:pulpcore_database)) && \
+            file_exists?('/var/lib/pgsql/data')
```
[the remark this answers is not in the document]
Comment (evgeni) [reply definitions/checks/disk/available_space_postgresql13.rb:9]: No idea 😁
Copied from old check

[the remark this answers is not in the document]
Comment (evgeni) [reply definitions/checks/disk/available_space_postgresql13.rb]: Ah, it was to check that the DB is local. Now better.

File: definitions/checks/disk/available_space_postgresql13.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      end
+
+      def psql_13_available_space
+        io_obj = ForemanMaintain::Utils::Disk::IODevice.new(pgsql_dir(13))
+        io_obj.available_space
+      end
+
+      def warning_message
+        "PostgreSQL will be upgraded from 12 to 13. \n" \
+        "During the upgrade a backup is created in /var/lib/pgsql/data-old and requires " \
```
Comment (ehelms) [inline definitions/checks/disk/available_space_postgresql13.rb:37]: Is the backup removed on successful upgrade or does it persist?

Comment (evgeni) [reply definitions/checks/disk/available_space_postgresql13.rb:37]: It persists and the user has to remove it

Comment (ehelms) [reply definitions/checks/disk/available_space_postgresql13.rb:37]: This is the same tactic we took before?

File: definitions/checks/disk/available_space_postgresql13.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      metadata do
+        label :available_space_for_postgresql13
+        description 'Check to make sure PostgreSQL 13 work directory has enough space for upgrade'
+        confine do
+          feature(:instance).postgresql_local?
+        end
+      end
+
+      def run
+        assert(psql_13_available_space >= psql_12_consumed_space, warning_message, :warn => true)
```
Comment (ehelms) [inline definitions/checks/disk/available_space_postgresql13.rb:13]: I am worried about warning being strong enough here. If there is not enough space, the upgrade will out right fail won't it?

File: definitions/checks/disk/available_space_postgresql13.rb (hunk 2 above)
Comment (evgeni) [reply definitions/checks/disk/available_space_postgresql13.rb:37]: That's what my memory says, yeah

(At least I recall having to special case el7 systems with /var/lib/pgsql still containing psql9 data in leapp)

File: definitions/checks/disk/available_space_postgresql13.rb (hunk 3 above)
Comment (evgeni) [reply definitions/checks/disk/available_space_postgresql13.rb:13]: Yeah, will change.

**Review by ehelms — APPROVED** [review]

**Pull request merged**